tests/unit/ml_handlers/test_langchain.py

```diff
@@ -0,0 +1,47 @@
+import os
+import pytest
+import pandas as pd
+from unit.ml_handlers.base_ml_test import BaseMLAPITest
+
+OPENAI_API_KEY = os.environ.get("OPENAI_API_KEY")
+
+
+@pytest.mark.skipif(OPENAI_API_KEY is None, reason='Missing OpenAI API key (OPENAI_API_KEY env variable)')
+class TestLangchain(BaseMLAPITest):
+    """Test Class for Langchain Integration Testing"""
+
+    def setup_method(self, method):
+        """Setup test environment, creating a project"""
+        super().setup_method()
+        self.run_sql("create database proj")
+
+    def test_conversational(self):
+        df = pd.DataFrame.from_dict({"question": [
```

Comment:
Depending on the feasibility to set this up, it'd be great to change the test so that it asks for e.g. the  available MindsDB tables, or the list of models. That'd be a task that a vanilla LLM engine would not be able to answer.